backend/endpoints/annotations/controller.js

```diff
@@ -47,22 +47,24 @@ export function postAnnotations(req, res) {
   // match the image ids whose annotations are being submitted
   .then((data) => {
     if (! data.length) {
-      throw new Error('unknown workload');
+      throw new Error('unknown workload: no stored workload');
     }
     const images = data[0].images;
     const storedIds = images.map(image => image.id);
     const submittedIds = req.body.images.map(image => image.id);
-    if (! isEqual(storedIds, submittedIds)) {
-      throw new Error('unknown workload');
+    if (! isEqual(storedIds.sort(), submittedIds.sort())) {
+      throw new Error('unknown workload: session workload of #s ${ submittedIds } did not match stored workload #s ${ storedIds }');
```

Comment:
the `'` should change to `` ` ``'s because you need back ticks for string substitution like this.